src/configurationProvider.ts

```diff
@@ -0,0 +1,74 @@
+import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { parse } from 'jsonc-parser';
+import { OmniSharpServer } from './omnisharp/server';
+import * as serverUtils from './omnisharp/utils';
+import { AssetGenerator, addTasksJsonIfNecessary, createLaunchConfiguration, createAttachConfiguration, containsDotNetCoreProjects} from './assets';
+import { WorkspaceInformationResponse } from './omnisharp/protocol';
+
+export class CSharpConfigurationProvider implements vscode.DebugConfigurationProvider {
+    private server: OmniSharpServer;
+
+    public constructor(server: OmniSharpServer) {
+        this.server = server;
+    }
+
+    /**
+     * TODO: Remove function. 
+     * 
+     * Note: serverUtils.requestWorkspaceInformation only retrieves one project. Therefore, generator will be incorrect for all folders
+     * except the first in a workspace. 
+     */
+    private checkWorkspaceInformationMatchesWorkspaceFolder(folder: vscode.WorkspaceFolder, info: WorkspaceInformationResponse): boolean {
+        return info.MsBuild.SolutionPath === folder.uri.fsPath;
```

Comment:
Note: It is entirely possible for `MsBuild` to be null.

Reply:
Other than this, the change looks good to me. Note that this can be null when the project is a `project.json` file.